pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -425,7 +425,7 @@ CompletableFuture<MessageId> internalSendWithTxnAsync(Message<?> message, Transa
      * @param payload
      * @return a new payload
      */
-    private ByteBuf applyCompression(ByteBuf payload) {
+    protected ByteBuf applyCompression(ByteBuf payload) {
```

Comment:
instead of opening the visibility of this method, maybe it is better to put here in ProducerImpl a method that creates the OpSendMsg.
I am not sure it is the better way, but maybe you take try to see if the code will look better

Reply:
It will not make the code look better now I think. 

Because publishing one message in batch and publish with batch disable will call different method.  

 As the new pushed commit `Fix out of order problem`  in this PR,  we should call `OpSendMsg#create(List<MessageImpl<?>>, ByteBufPair, long, SendCallback)` to create OpSendMsg, instead of using `OpSendMsg#create(MessageImpl<?>,  ByteBufPair,  long,  SendCallback)`.